modules/heat_conduction/src/constraints/ModularGapConductanceConstraint.C

```diff
@@ -370,6 +388,9 @@ ModularGapConductanceConstraint::computeQpResidual(Moose::MortarType
       // Ensure energy balance for non-flat (non-PLATE) general geometries when using radiation
       _surface_integration_factor = computeSurfaceIntegrationFactor();
 
+      // Set the value of the normal contact pressure for the user objects that require it
+      _normal_pressure = _contact_pressure[_qp];
```

Comment:
@sapitts sorry that I'm seeing this so late, but why can't the GapFlux model couple to the pressure variable directly? Why is it passed through the constraint?

Reply:
Oh, because it lives on the lower-d block and not the interfaces...

Reply:
Yeah, it wasn't pulling the right value of the variable on the lower dimensional block.

Reply:
Yeah, I pushed a commit to couple in the GapFluxModel. That seems to be a better design and I hope we can make it work. Currently it exodiffs.

Reply:
Ok, this was a tricky one. The difference between getting the contact pressure in the UO vs. the Constraint was that the  Constraint operated on the displaced mesh and the OU operated on the undisplaced mesh. We should probably default the flux models to the displaced mesh and add a check that constraint and flux models operate on the same mesh.